src/transformers/naive_split.cc

```diff
@@ -78,9 +75,19 @@ namespace detail {
 		assert(std::distance(original->get_dependents().begin(), original->get_dependents().end()) == 0);
 
 		chunk<3> full_chunk{tsk.get_global_offset(), tsk.get_global_size(), tsk.get_global_size()};
-		auto chunks = split_equal(full_chunk, tsk.get_granularity(), num_chunks, tsk.get_dimensions());
+		const auto chunks = split_equal(full_chunk, tsk.get_granularity(), num_chunks, tsk.get_dimensions());
+		assert(chunks.size() <= num_chunks); // We may have created less than requested
```

Comment:
Also, `chunks.size()` must not be 0 here.

Reply:
Done!